server/aesdsocket.c

```diff
@@ -143,6 +145,57 @@ void *get_in_addr(struct sockaddr *sa) {
     return &(((struct sockaddr_in6 *) sa)->sin6_addr);
 }
 
+bool check_for_ioctl_cmd(char *temp_file, uint32_t *cmd_num, uint32_t *cmd_offset) {
+    char pattern[] = "(AESDCHAR_IOCSEEKTO:)([0-9]+),([0-9]+)";
+    char cmd_num_str[100];
+    char cmd_offset_str[100];
+    regex_t regex;
+    regmatch_t matches[4];
+    bool result;
+
+    // Compile the regular expression
+    int reti = regcomp(&regex, pattern, REG_EXTENDED);
+    if (reti) {
+        fprintf(stderr, "Regex compilation failed\n");
+        exit(EXIT_FAILURE);
```

Comment:
Killing the process here is a bit harsh.  Also you could move the regex compile stage into init and only do this once.